Added script that relaunches kmonad on crash

PockyBum522 opened the pull request:
This powershell script launches kmonad, then watches for the error message "Encountered error in KeySource: Cannot translate from windows keycode" which means the kmonad process has crashed.

If it sees this, it immediately kills the existing kmonad process and then launches a new one, again watching for a crash.

It also handles Ctrl + C just fine and will kill any existing kmonad processes before exiting as long as the matching pattern is correctly in the "finally" block at the end of this file.

I am submitting this pull request due to slotThe's comment in this thread which relates to a linux version of this script: https://github.com/kmonad/kmonad/issues/432

PockyBum522 committed:
windows: Add script that relaunches kmonad on crash

This powershell script launches kmonad, then watches for the error
message "Encountered error in KeySource: Cannot translate from windows
keycode" which means the kmonad process has crashed.

If it sees this, it immediately kills the existing kmonad process and
then launches a new one, again watching for a crash.

It also handles C-c just fine and will kill any existing kmonad
processes before exiting as long as the matching pattern is correctly in
the "finally" block at the end of this file.

Calling the script works like this:

  .\relaunch_kmonad_after_keycode_error.ps1 -binary "C:\Full\Path\To\Your\kmonad_binary.exe" -config "C:\Full\Path\To\Your_Kmonad_Config.kbd"

Both arguments are REQUIRED and both MUST BE surrounded by double
quotes.

slotThe commented on scripts/relaunch_kmonad_after_keycode_error.ps1:
```diff
@@ -0,0 +1,115 @@
+    
+        $blockToRun = 
+        {        
+            $kmonadPath = "D:\Kmonad-Windows\kmonad-key-repeat-fixed.exe"
+            $kbdConfigPath = '"D:\source\repos\linux-files\configuration\kmonad\david_default.kbd"'
```
Perhaps we should set these to definitely non-existent files, so this just results in an error if the user forgot to set these. Also, wouldn't it make more sense to make these CLI arguments?

PockyBum522 replied:
I will make it more clear that those are just the kmonad binary, and the user's config file. You are absolutely correct.

Making them arguments is also a good idea. I will do that.

PockyBum522 commented:
* Changed example paths for binary and config to obviously nonexistent paths, as suggested.

* Changed paths from being hardcoded in script to parameters when calling the script

* Added checks that parameters were passed when calling the script. If either parameter is not present or empty or whitespace, then an example is written to the console with information on what needs to be passed and how to do so.

slotThe commented:
Thank you!
